File: common/src/main/java/dev/vini2003/hammer/core/registry/server/HCEvents.java
package dev.vini2003.hammer.core.registry.server;

import dev.architectury.event.events.common.LifecycleEvent;
import dev.vini2003.hammer.core.api.client.util.InstanceUtil;

public class HCEvents {
	public static void init() {
		LifecycleEvent.SERVER_STARTED.register((server) -> {
			InstanceUtil.setServerSupplier(() -> server);
		});
	}
}

File: hammer-chat/src/main/kotlin/dev/vini2003/hammer/chat/registry/common/HCEvents.kt
package dev.vini2003.hammer.chat.registry.common

import dev.vini2003.hammer.chat.listener.SyncGlobalChatListener
import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents

object HCEvents {
	fun init() {
		ServerPlayConnectionEvents.JOIN.register(SyncGlobalChatListener)
	}
}